cli/lib/build.ts

```diff
@@ -353,18 +382,35 @@ export default async (options: { local?: boolean; image?: string; enableSourceMa
         }
         const requireBuildList = await getRequireBuildModules(image);
         logInfo(`Will launch a partial build based on image ${image}`);
-        rebuildList = await restoreFromDockerImage(image, requireBuildList);
+        rebuildList = await restoreFromDockerImage(image, requireBuildList, dataEngineerInfo);
       }
       await checkReInstall();
     }
 
-    await buildModules(enableSourceMap, rebuildList, `${online}`);
+    await buildModules(enableSourceMap, rebuildList, `${online}`, dataEngineerInfo);
 
     generateVersion();
 
     if (rebuildList.includes('shell')) {
       localIcon();
     }
+
+    externalModules.forEach(({ nginx, env }) => {
+      if (nginx) {
+        const nginxPath = path.resolve(process.cwd(), 'nginx.conf.template');
+        const nginxContent = fs.readFileSync(nginxPath).toString('utf8').split(EOL);
+        nginxContent.splice(-1, 0, nginx);
```

Comment:
Should add some comment as insert mark, because nginx route match has order.

Reply:
from my perspective，I think place the fdp proxy in the end of nginx file should have no impact on the route logic.